src/contexts/PreferencesContext.tsx

```diff
@@ -265,91 +268,79 @@ export const PreferencesProvider = ({
-      key,
-      { type: 'zone', name: item.name },
-      zonePreferenceMap
-    ) as Record<string, ZonePreference>;
-
     try {
+      const key = makeMapKey('zone', item.name);
+      const updatedZonePreferenceMap = updatePreferenceList(
+        key,
+        { type: 'zone', name: item.name },
+        zonePreferenceMap
+      ) as Record<string, ZonePreference>;  
       await savePreferencesToBackend(
         'zone',
         Object.values(updatedZonePreferenceMap)
       );
       updateLocalZonePreferenceStates(updatedZonePreferenceMap);
     } catch (error) {
-      console.error('Error updating zone favorites:', error);
+      console.error('Error in handleZoneFavoriteChange:', error);
     }
   }
 
   async function handleFileSharePathFavoriteChange(item: FileSharePath) {
-    const key = makeMapKey('fsp', item.name);
-    const updatedFileSharePathMap = updatePreferenceList(
-      key,
-      { type: 'fsp', name: item.name },
-      fileSharePathPreferenceMap
-    ) as Record<string, FileSharePathPreference>;
-
     try {
+      const key = makeMapKey('fsp', item.name);
+      const updatedFileSharePathMap = updatePreferenceList(
+        key,
+        { type: 'fsp', name: item.name },
+        fileSharePathPreferenceMap
+      ) as Record<string, FileSharePathPreference>;
       await savePreferencesToBackend(
         'fileSharePath',
         Object.values(updatedFileSharePathMap)
       );
       updateLocalFspPreferenceStates(updatedFileSharePathMap);
     } catch (error) {
-      console.error('Error updating file share path favorites:', error);
+      console.error('Error in handleFileSharePathFavoriteChange:', error);
     }
   }
 
   async function handleFolderFavoriteChange(item: FolderFavorite) {
-    const folderPrefKey = makeMapKey(
-      'folder',
-      `${item.fsp.name}_${item.folderPath}`
-    );
-    const updatedFolderMap = updatePreferenceList(
-      folderPrefKey,
-      { type: 'folder', folderPath: item.folderPath, fspName: item.fsp.name },
-      folderPreferenceMap
-    ) as Record<string, FolderPreference>;
-
     try {
+      const folderPrefKey = makeMapKey(
+        'folder',
+        `${item.fsp.name}_${item.folderPath}`
+      );
+      const updatedFolderMap = updatePreferenceList(
+        folderPrefKey,
+        { type: 'folder', folderPath: item.folderPath, fspName: item.fsp.name },
+        folderPreferenceMap
+      ) as Record<string, FolderPreference>;
       await savePreferencesToBackend('folder', Object.values(updatedFolderMap));
       updateLocalFolderPreferenceStates(updatedFolderMap);
     } catch (error) {
-      console.error('Error updating folder favorites:', error);
+      console.error('Error in handleFolderFavoriteChange:', error);
     }
   }
 
   async function handleFavoriteChange(
     item: Zone | FileSharePath | FolderFavorite,
     type: string
   ) {
-    switch (type) {
-      case 'zone':
-        try {
+    try {
```

Comment:
Moved the try block to the outside so it only has to be done once instead of for each case.